portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -25,7 +33,54 @@
 public class DataServlet extends HttpServlet {
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("comment");
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    ArrayList<Comment> comments = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      String message = (String) entity.getProperty("message");
+      Comment comment = new Comment(message);
+      comments.add(comment);
+    }
+
+    response.setContentType("application/json;");
+    response.getWriter().println(convertToJson(comments));
+  }
+
+  private String convertToJson(ArrayList<Comment> cmts) {
+    Gson gson = new Gson();
+    String json = gson.toJson(cmts);
+    return json;
```

Comment:
To improve readability, it's better to organize your code to align public methods together, and private methods together.